appeals/web/src/server/lib/validators/date-input.validator.js

```diff
@@ -14,119 +14,69 @@ export const createDateInputFieldsValidator = (
 	messageFieldNamePrefix = 'date',
 	dayFieldName = '-day',
 	monthFieldName = '-month',
-	yearFieldName = '-year'
-) =>
-	createValidator(
-		body(`${fieldNamePrefix}${dayFieldName}`)
-			.exists()
-			.withMessage(
-				capitalize(
-					`${(messageFieldNamePrefix && messageFieldNamePrefix + ' d') || 'D'}ay cannot be empty`
-				)
-			)
-			.bail()
-			.trim()
-			.notEmpty()
-			.withMessage(
-				capitalize(
-					`${(messageFieldNamePrefix && messageFieldNamePrefix + ' d') || 'D'}ay cannot be empty`
-				)
-			)
-			.bail()
+	yearFieldName = '-year',
+	bodyScope = ''
+) => {
+	const _day = bodyScope ? `[${dayFieldName}]` : dayFieldName;
+	const _month = bodyScope ? `[${monthFieldName}]` : monthFieldName;
+	const _year = bodyScope ? `[${yearFieldName}]` : yearFieldName;
+
+	return createValidator(
+		body(bodyScope).custom((value) => {
+			const day = value[`${fieldNamePrefix}${dayFieldName}`];
+			const month = value[`${fieldNamePrefix}${monthFieldName}`];
+			const year = value[`${fieldNamePrefix}${yearFieldName}`];
+
+			if (day && month && year) {
+				return true;
+			}
+
+			let missingParts = [];
```

Comment:
Very pedantic, but this could be `const`

Reply:
I tend to like to mark arrays as let if I'm going to push to them. As much as you don't need to, it lets someone reading it know the value will change.